File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusElectAccept.java
package com.github.dedis.popstellar.model.network.method.message.data.consensus;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.network.method.message.data.*;
import com.github.dedis.popstellar.model.objects.security.MessageID;
import com.google.gson.annotations.SerializedName;

@Immutable
public final class ConsensusElectAccept extends Data {

  @SerializedName("instance_id")
  private final String instanceId;

  @SerializedName("message_id")
  private final MessageID messageId;

  private final boolean accept;

  /**
   * Constructor for a data Elect_Accept
   *
   * @param instanceId unique id of the consensus instance
   * @param messageId message id of the Elect message
   * @param accept true if the node agrees with the proposal
   */
  public ConsensusElectAccept(String instanceId, MessageID messageId, boolean accept) {
    this.instanceId = instanceId;
    this.messageId = messageId;
    this.accept = accept;
  }

  public String getInstanceId() {
    return instanceId;
  }

  public MessageID getMessageId() {
    return messageId;
  }

  public boolean isAccept() {
    return accept;
  }

  @Override
  public String getObject() {
    return Objects.CONSENSUS.getObject();
  }

  @Override
  public String getAction() {
    return Action.ELECT_ACCEPT.getAction();
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(instanceId, messageId, accept);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ConsensusElectAccept that = (ConsensusElectAccept) o;

    return accept == that.accept
        && java.util.Objects.equals(instanceId, that.instanceId)
        && java.util.Objects.equals(messageId, that.messageId);
  }

  @Override
  public String toString() {
    return String.format(
        "ConsensusElectAccept{instance_id='%s', message_id='%s', accept=%b}",
        instanceId, messageId.getEncoded(), accept);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusElectAccept.kt
package com.github.dedis.popstellar.model.network.method.message.data.consensus

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.network.method.message.data.Action
import com.github.dedis.popstellar.model.network.method.message.data.Data
import com.github.dedis.popstellar.model.network.method.message.data.Objects
import com.github.dedis.popstellar.model.objects.security.MessageID
import com.google.gson.annotations.SerializedName

@Immutable
class ConsensusElectAccept
/**
 * Constructor for a data Elect_Accept
 *
 * @param instanceId unique id of the consensus instance
 * @param messageId message id of the Elect message
 * @param isAccept true if the node agrees with the proposal
 */
(
  @field:SerializedName("instance_id") val instanceId: String,
  @field:SerializedName("message_id") val messageId: MessageID,
  @field:SerializedName("accept") val isAccept: Boolean
) : Data() {

  override val `object`: String = Objects.CONSENSUS.`object`
  override val action: String = Action.ELECT_ACCEPT.action

  override fun hashCode(): Int {
    return java.util.Objects.hash(instanceId, messageId, isAccept)
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as ConsensusElectAccept
    return isAccept == that.isAccept && instanceId == that.instanceId && messageId == that.messageId
  }

  override fun toString(): String {
    return "ConsensusElectAccept{instance_id='$instanceId', message_id='${messageId.encoded}', accept=$isAccept}"
  }
}
